src/types/spend_bundle.py

```diff
@@ -46,6 +46,9 @@ def announcements(self) -> List[Announcement]:
             items.extend(coin_solution.announcements())
         return items
 
+    def announcement_names(self) -> List[bytes32]:
+        return [cs.announcement_names() for cs in self.coin_solutions]
+
```

Comment:
Are there any tests for this? It looks to me that this is going to return a list of lists, many likely empty, which is probably not what you want. Consider `.extend`.